source/screenCurtain.py

```diff
@@ -0,0 +1,234 @@
+
+	This class should be treated as a singleton:
+	There should only ever be a single object created from this class at a time.
+	"""
+
+	def __init__(self):
+		"""Initializer."""
+		super().__init__()
+		self._settings: _ScreenCurtainSettings = cast(_ScreenCurtainSettings, config.conf["screenCurtain"])
+		self._enabled: bool = False
+		if self.settings["enabled"]:
+			self.enable()
+
+	@property
+	def settings(self) -> _ScreenCurtainSettings:
+		"""The settings for the Screen Curtain."""
+		return self._settings
+
+	@property
+	def enabled(self) -> bool:
+		"""Whether the Screen Curtain is currently enabled."""
+		return self._enabled
+
+	def enable(self, *, persist=False) -> None:
+		"""Enables the screen curtain.
+
+		This method is idempotent.
+
+		:param persist: Whether to write that the Screen Curtain has been enabled to config, defaults to False
+		:raises RuntimeError: On failure to activate the Screen Curtain
+		"""
+		if self._enabled:
+			log.debug("ScreenCurtain is already enabled.")
+			return
+		log.debug("Enabling ScreenCurtain")
+		magnification.MagInitialize()
+		try:
+			magnification.MagSetFullscreenColorEffect(TRANSFORM_BLACK)
+			magnification.MagShowSystemCursor(False)
+			if not isScreenFullyBlack():
+				raise RuntimeError("Screen is not black.")
+		except Exception as e:
+			magnification.MagUninitialize()
+			raise e
+		self._enabled = True
+		if persist:
+			self.settings["enabled"] = True
+		if self.settings["playToggleSounds"]:
+			try:
+				nvwave.playWaveFile(os.path.join(globalVars.appDir, "waves", "screenCurtainOn.wav"))
+			except Exception:
+				log.exception()
+
+	def disable(self, *, persist: bool = True):
+		"""Disables the Screen Curtain.
+
+		This method is idempotent.
+
+		:param persist: Whether to store that the Screen Curtain has been disabled to config, defaults to True
+		"""
+		if not self._enabled:
+			log.debug("ScreenCurtain is already disabled")
+			return
+		log.debug("Disabling ScreenCurtain")
+		magnification.MagShowSystemCursor(True)
+		magnification.MagUninitialize()
+		self._enabled = False
+		if persist:
+			self.settings["enabled"] = False
+		if self.settings["playToggleSounds"]:
+			try:
+				nvwave.playWaveFile(os.path.join(globalVars.appDir, "waves", "screenCurtainOff.wav"))
+			except Exception:
+				log.exception()
+
+	def __del__(self):
+		"""Custom deleter that disables the Screen Curtain if necessary when this object is garbage collected."""
+		if self._enabled:
+			self.disable(persist=False)
+		if hasattr(super(), "__del__"):
+			super().__del__(self)
+
+
+screenCurtain: _ScreenCurtain | None = None
+"""Global Screen Curtain controller."""
+
+
+def initialize():
+	"""Initialize theScreen Curtain."""
+	global screenCurtain
+	if screenCurtain is None:
+		screenCurtain = _ScreenCurtain()
+
+
+def terminate():
+	"""Terminate the Screen Curtain."""
+	global screenCurtain
+	if screenCurtain is not None:
+		screenCurtain.disable(persist=False)
+		screenCurtain = None
```

Comment:
could we consider making this a module and just keeping this part in the __init__.py?